pkg/io/jsonio/unmarshal.go

```diff
@@ -147,22 +143,27 @@ func replaceResourceName(distinctNames map[string]ir.ID, ambiguousNames map[stri
 	return "", fmt.Errorf("unknown resource name %s (resource type: %q)", resourceName, resourceType)
 }
 
-// inverseMapToFullyQualifiedName returns two maps: one from a name to a fully qualified name,
+// detectDistinctAndAmbiguousNames returns two maps: one from a name to a fully qualified name,
 // and the second is a set of ambiguous names
-func inverseMapToFullyQualifiedName[T ir.Named](m map[ir.ID]T) (distinctNames map[string]ir.ID, ambiguousNames map[string]struct{}) {
+func detectDistinctAndAmbiguousNames[T any](m map[ir.ID]T) (distinctNames map[string]ir.ID, ambiguousNames map[string]struct{}) {
 	ambiguousNames = make(map[string]struct{})
 	distinctNames = make(map[string]ir.ID)
 
-	for fullElementName, element := range m {
-		elementName := element.Name()
-		if _, ok := ambiguousNames[elementName]; ok {
-			continue
-		}
-		if _, ok := distinctNames[elementName]; !ok {
-			distinctNames[elementName] = fullElementName
-		} else {
-			delete(distinctNames, elementName)
-			ambiguousNames[elementName] = struct{}{}
+	for fullElementName := range m {
+		distinctNames[fullElementName] = fullElementName
+		for i, c := range fullElementName {
+			if c == '/' && i+1 < len(fullElementName) {
+				currName := fullElementName[i+1:]
+				if _, ok := ambiguousNames[currName]; ok {
+					break
```

Comment:
```suggestion
					continue
```
If there are two NIFs, `vpc1/vsi1/nif1` and `vpc2/vsi1/nif1`, `nif1` currently is not marked as ambiguous (or distinct), just `vsi1/nif1`.